src/core/border_router/routing_manager.cpp

```diff
@@ -204,7 +213,57 @@ void RoutingManager::GenerateOmrPrefix(void)
     LogInfo("Generated OMR prefix: %s", mLocalOmrPrefix.ToString().AsCString());
 }
 
+const char    RoutingManager::kWellKnownIpv4OnlyName[]     = "ipv4only.arpa";
+const uint8_t RoutingManager::kWellKnownIpv4OnlyAddress1[] = {192, 0, 0, 170};
+const uint8_t RoutingManager::kWellKnownIpv4OnlyAddress2[] = {192, 0, 0, 171};
+
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+void RoutingManager::RequestInfraIfNat64Prefix(void)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsInitialized() && mInfraIf.IsRunning(), error = kErrorInvalidState);
+
+    LogInfo("Requesting infrastructure NAT64 prefix from %s on %s", kWellKnownIpv4OnlyName,
+            mInfraIf.ToString().AsCString());
+    error = mInfraIf.RequestHostAddress(kWellKnownIpv4OnlyName);
+
+exit:
+    if (error != kErrorNone)
+    {
+        LogWarn("Failed to request upstream NAT64 prefix on %s: %s", mInfraIf.ToString().AsCString(),
```

Comment:
```suggestion
        LogWarn("Failed to request AIL NAT64 prefix on %s: %s", mInfraIf.ToString().AsCString(),
```
AIL stands for adjacent infrastructure link